mtg_card_identifier/CardScannerFile.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  CardScannerFile.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import UIKit
+import VisionKit
+
+struct CardScanner: UIViewControllerRepresentable {
+    
+    //    @Binding var startScanning: Bool
+    @Binding var scanedCard: [String]
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(self)
+    }
+    
+    func makeUIViewController(context: Context) -> DataScannerViewController {
+        
+        let vc = DataScannerViewController(
+            recognizedDataTypes: [.text()],
+            qualityLevel: .balanced,
+            recognizesMultipleItems: true,
+            isHighFrameRateTrackingEnabled: true
+            //            isGuidanceEnabled: true,
+//            Don't know if we want that to work but it's basically done
+//            isHighlightingEnabled: true
+        )
+        let partialTransparentView = PartialTransparentView(cutout: CGRect(x: 31, y: 75, width: 350, height: 470))
+        partialTransparentView.translatesAutoresizingMaskIntoConstraints = true
+        partialTransparentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        partialTransparentView.frame = vc.overlayContainerView.bounds
+        
+        vc.view.addSubview(partialTransparentView)
+        vc.regionOfInterest = CGRect(x: 31, y: 75, width: 350, height: 430)
+        vc.delegate = context.coordinator
+        
+        return vc
+    }
+    
+    func updateUIViewController(_ uiViewController: DataScannerViewController, context: Context) {
+        do {
+            try uiViewController.startScanning()
+            //            I don't know how safe this is
+        } catch {
+            
+            print("Error: \(error.localizedDescription)")
+        }
+    }
+    
+    @MainActor
+    class Coordinator: NSObject, DataScannerViewControllerDelegate {
+        var parent: CardScanner
+        
+        init(_ parent: CardScanner) {
+            self.parent = parent
```

Comment:
The only reason we need the parent in this object is to grab the Binding off of it. I might just store that Binding rather than hold on to the parent. That feels a bit cleaner to me, but others might disagree.